test/realtime/extensions/cdc_rls/replication_poller_test.exs

```diff
@@ -287,14 +287,21 @@ defmodule ReplicationPollerTest do
   end
 
   describe "slot_name_suffix" do
-    test "when no SLOT_NAME_SUFFIX" do
-      System.delete_env("SLOT_NAME_SUFFIX")
-      assert Poller.slot_name_suffix() == ""
+    setup do
+      slot_name_suffix = Application.get_env(:realtime, :slot_name_suffix)
+
+      on_exit(fn -> Application.put_env(:realtime, :slot_name_suffix, slot_name_suffix) end)
```

Comment:
why does it need to put this key on the exit?

Reply:
resetting the env to be equal to what it was in the start since we're changing application setup